base/src/types/mod.rs

```diff
@@ -558,6 +589,16 @@ pub enum Type<Id, T = ArcType<Id>> {
     /// when the binding is refered to.
     Generic(#[cfg_attr(feature = "serde_derive", serde(state))] Generic<Id>),
     Alias(#[cfg_attr(feature = "serde_derive", serde(state))] AliasRef<Id, T>),
+    Skolem(#[cfg_attr(feature = "serde_derive", serde(state))] Skolem<Id>),
```

Comment:
So we didn't end up gaining a `Type::Abs` variant? How do we represent `type Id a = a`? Afaik, this should become `Id : Type -> Type = \a : Type -> a`...

Reply:
`Type::Alias` work fine to represent that case so there is no specific to have a specific `Type::Abs` case.

In core expressions lambdas actually gets desugared this `\x -> x` => `let fun x = x in func`.